core/src/main/java/hudson/model/Queue.java

```diff
@@ -1522,8 +1532,44 @@ public void maintain() {
     private @CheckForNull Runnable makeBuildable(final BuildableItem p) {
         if (p.task instanceof FlyweightTask) {
             if (!isBlockedByShutdown(p.task)) {
+
+                Runnable runnable = makeFlyWeightTaskBuildable(p);
+
+                if(runnable!=null){
+                    return runnable;
+                }
+
+                //this is to solve JENKINS-30084: the task has to be buildable to force the provisioning of nodes.
+                //if the execution gets here, it means the task could not be scheduled since the node
+                //the task is supposed to run on is offline or not available.
+                return new Runnable() {
+                    @Override public void run() {
+                        //the flyweighttask enters the buildables queue and will ask Jenkins to provision a node
+                        p.enter(Queue.this);
+                    }
+                };
+            }
+            // if the execution gets here, it means the task is blocked by shutdown and null is returned.
+            return null;
+        } else { // regular heavyweight task
+            return new Runnable() {
+                @Override public void run() {
```

Comment:
why in some places you write annotation on line before code, and in other places - inline?
Please write on a separate line annotations